multimedia/ffmpeg4/Makefile

```diff
@@ -0,0 +1,342 @@
+	$(if $(CONFIG_ffmpeg4_ENC_MPEG4),--enable-encoder=mpeg4,) \
+	$(if $(CONFIG_ffmpeg4_ENC_PNG),--enable-encoder=png,) \
+	$(if $(CONFIG_ffmpeg4_LIB_FDK-AAC),--enable-decoder=libfdk_aac --enable-encoder=libfdk_aac --enable-libfdk-aac --enable-nonfree, --disable-libfdk-aac) \
+	$(if $(CONFIG_ffmpeg4_LIB_LAME),--enable-encoder=libmp3lame --enable-libmp3lame,--disable-libmp3lame) \
+	$(if $(CONFIG_ffmpeg4_LIB_MBEDTLS),--enable-mbedtls --enable-version3,--disable-mbedtls) \
+	$(if $(CONFIG_ffmpeg4_LIB_OPENSSL),--enable-openssl --enable-nonfree,--disable-openssl) \
+	$(if $(CONFIG_ffmpeg4_LIB_LIBOPUS),--enable-decoder=libopus --enable-encoder=libopus --enable-libopus,--disable-libopus) \
+	$(if $(CONFIG_ffmpeg4_LIB_LIBVORBIS),--enable-decoder=libvorbis --enable-encoder=libvorbis --enable-libvorbis,--disable-libvorbis) \
+	$(if $(CONFIG_ffmpeg4_LIB_LIBX264),--enable-encoder=libx264 --enable-encoder=libx264rgb --enable-libx264 --enable-gpl,--disable-libx264) \
+	$(if $(CONFIG_ffmpeg4_OPT_PERFORMANCE),,--enable-small) \
+
+### Platform specific fixes
+
+### ARM (XScale)
+### Note: No need to include instructions that aren't supported
+ifeq ($(CONFIG_SOFT_FLOAT),y)
+  ifeq ($(call qstrip,$(CONFIG_CPU_TYPE)),xscale)
+    CONFIGURE_ARGS += --disable-armv6 --disable-armv6t2 --disable-vfp \
+    --disable-neon --disable-fast-unaligned
+  endif
+endif
+
+### MIPS 32-bit and 64-bit
+### Note: FFmpeg does support MIPS32R2 optimization however it uses opcodes
+### that aren't supported hence --cpu=generic otherwise the build will fail.
+### Enabling assembly generates "invalid operands" using GCC 8.3.0 and
+### binutils 2.32, additionally MIPS16 breaks LTO
+ifeq ($(CONFIG_SOFT_FLOAT),y)
+  ifneq (,$(filter $(call qstrip,$(CONFIG_CPU_TYPE)), mips32 24kc 34kc 74kc octeonplus))
+    CONFIGURE_ARGS += --disable-mipsfpu --disable-mipsdsp \
+	--disable-mipsdspr2 --disable-fast-unaligned --disable-msa \
+	--disable-mmi --disable-asm --cpu=generic
+  endif
+endif
+
+ifneq ($(CONFIG_USE_MIPS16),y)
+  CONFIGURE_ARGS += --enable-lto
+endif
+
+### PowerPC apm821xx
+### Note: No need to include instructions that aren't supported
+ifeq ($(call qstrip,$(CONFIG_CPU_TYPE)),464fp)
+  CONFIGURE_ARGS += --disable-altivec --disable-vsx
+endif
+
+### x86 / x86-64
+### Note: FFmpeg blindly assumes that NASM is available
+### NASM produces broken files for i386 subtargets except for pentium4
+### For whatever reason compilation using GCC 8.3.0 and binutils 2.32 fails in
+### libswscale/x86/yuv2rgb_template.c with the following message:
+### "error: 'asm' operand has impossible constraints" on x86 (32-bit)
+### with assembly enabled. For now just disable assembly to bypass this issue.
+### Some assembly files for x86 are licensed under
+### GNU General Public License v2.0 or later
+### https://github.com/FFmpeg/FFmpeg/blob/master/LICENSE.md
+
+ifeq ($(CONFIG_TARGET_x86),y)
+  CONFIGURE_ARGS += --enable-gpl --disable-asm
+  ifneq ($(CONFIG_NASM),y)
```

Comment:
I fixed this a different way in the other ffmpeg package because of upstream. Should match that.

Reply:
Link?

Reply:
https://github.com/openwrt/packages/commit/013162d07968738bf837370a91536d35b8f780e7